src/modeler/AutoRest.Swagger/Validation/Extensions/MutabilityWithReadOnlyRule.cs

```diff
@@ -70,5 +71,10 @@ private bool ValidateMutabilityValuesWithReadOnlyProperty(object mutable, RuleCo
 
             return isValid;
         }
+
+        /// <summary>
+        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// </summary>
+        public override Category Severity => Category.Error;
```

Comment:
do you mind reordering code a bit, and as you're adding ID and category, place them at the beginning, with this one next, then the message template, and then the rest of the methods?